### PR TITLE
enh(conf) handle new centengine logger in configuration

### PR DESCRIPTION
**Pull request by a-launois**
## Description
Handle logger V2 in centengine configuration (form, export, clapi)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: lib/perl/centreon/script/centreonSyncArchives.pm [hunk 1]
```diff
@@ -61,10 +61,12 @@ sub run {
                                         password => $self->{centreon_config}->{db_passwd},
                                         force => 0,
                                         logger => $self->{logger});
-    my ($status, $sth) = $cdb->query("SELECT nagios_server.id, nagios_server.ns_ip_address, nagios_server.ssh_port, cfg_nagios.log_archive_path FROM nagios_server, cfg_nagios 
+    my ($status, $sth) = $cdb->query("SELECT nagios_server.id, nagios_server.ns_ip_address, nagios_server.ssh_port, cfg_nagios.log_file FROM nagios_server, cfg_nagios
                                       WHERE nagios_server.ns_activate = '1' AND nagios_server.localhost = '0' AND nagios_server.id = cfg_nagios.nagios_server_id");
     die("Error SQL Quit") if ($status == -1);
     while ((my $data = $sth->fetchrow_hashref())) {
+        $data->{log_archive_path} = $data->{log_file};
+        $data->{log_archive_path} =~ s/(.*)\/.*/$1\/archives\//;
```
Comment (kduret) [inline lib/perl/centreon/script/centreonSyncArchives.pm:69]:
```suggestion
        $data->{log_archive_path} =~ s#(.*)/.*#$1/archives/#;
```

File: src/CentreonRemote/Domain/Resources/DefaultConfig/CfgNagiosLogger.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?php
+
```
Comment (kduret) [inline src/CentreonRemote/Domain/Resources/DefaultConfig/CfgNagiosLogger.php:2]: missing license header

File: src/CentreonRemote/Domain/Resources/RemoteConfig/CfgNagiosLogger.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?php
+
```
Comment (kduret) [inline src/CentreonRemote/Domain/Resources/RemoteConfig/CfgNagiosLogger.php:2]: missing license header

File: www/class/centreon-clapi/centreonEngineCfg.class.php [hunk 4]
```diff
@@ -229,8 +246,13 @@ public function initUpdateParameters($parameters)
                 } else {
                     $params[2] = null;
                 }
+            } elseif ($params[1] === 'logger_version' && $params['2'] === 'log_v2_enabled') {
```
Comment (kduret) [inline www/class/centreon-clapi/centreonEngineCfg.class.php:249]:
```suggestion
            } elseif ($params[1] === 'logger_version' && $params[2] === 'log_v2_enabled') {
```

File: www/class/config-generate/engine.class.php [hunk 5]
```diff
@@ -344,6 +360,45 @@ private function getIntervalLength()
         $this->engine['interval_length'] = $this->stmt_interval_length->fetchAll(PDO::FETCH_COLUMN);
     }
 
+    /**
+     *  If log Vé enabled, set logger V2 configuration and unset logger legacy elements
```
Comment (kduret) [inline www/class/config-generate/engine.class.php:364]:
```suggestion
     *  If log V2 enabled, set logger V2 configuration and unset logger legacy elements
```

File: www/include/configuration/configNagios/formNagios.php [hunk 6]
```diff
@@ -52,8 +52,9 @@
 /*
  * Database retrieve information for Nagios
  */
-$nagios = array();
-$nagios_d = array();
+$nagios = [];
+$nagios_log_v1 = [];
+$nagios_log_v2 = [];
```
Comment (kduret) [inline www/include/configuration/configNagios/formNagios.php:57]: use camelcase

File: www/include/configuration/configNagios/formNagios.php [hunk 7]
```diff
@@ -64,10 +65,19 @@
...
     foreach ($tmp as $key => $value) {
-        $nagios_d["nagios_debug_level"][$value] = 1;
+        $nagios_log_v1["nagios_debug_level"][$value] = 1;
     }
+    // Log V2
+    if ($nagios['logger_version'] === 'log_v2_enabled') {
+        $statement = $pearDB->prepare("SELECT * FROM cfg_nagios_logger WHERE cfg_nagios_id = :nagiosId");
+        $statement->bindValue(':nagiosId', $nagiosId, \PDO::PARAM_INT);
+        $statement->execute();
+        $nagios_log_v2 = $statement->fetch();
```
Comment (kduret) [inline www/include/configuration/configNagios/formNagios.php:78]: need to check if fetch returns false

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -16793,3 +16793,24 @@ msgstr "Licence invalide ou expirée"
 
 msgid "Please contact your administrator"
 msgstr "Veuillez contacter votre administrateur"
+
```
Comment (a-launois) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16796]: Add translation for other languages

File: src/CentreonRemote/Domain/Resources/RemoteConfig/CfgNagiosLogger.php [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+/*
+ * Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
+ *
```
Comment (a-launois) [inline src/CentreonRemote/Domain/Resources/RemoteConfig/CfgNagiosLogger.php:5]: use Default CfgNagiosLogger

File: www/class/centreon-clapi/centreonEngineCfg.class.php [hunk 10]
```diff
@@ -454,4 +478,73 @@ protected function delBkModule($objectId, $brokerModule)
             }
         }
     }
+
+    /**
+     * @param int $nagiosId
+     */
+    public function insertRelations(int $nagiosId): void
```
Comment (a-launois) [inline www/class/centreon-clapi/centreonEngineCfg.class.php:485]:
```suggestion
    /**
     * This method is automatically called from centreonObject
     * @param int $nagiosId
     */
    public function insertRelations(int $nagiosId): void
```

File: www/class/centreon-clapi/centreonEngineCfg.class.php [hunk 11]
```diff
@@ -454,4 +478,73 @@ protected function delBkModule($objectId, $brokerModule)
...
+        $statement->bindValue(':cfg_nagios_id', $nagiosId, \PDO::PARAM_INT);
+        $statement->execute();
+
+        if ($statement->fetch()) {
+            $statement = $this->db->prepare(
+                "UPDATE cfg_nagios_logger SET {$params[1]} = :paramValue WHERE cfg_nagios_id = :cfg_nagios_id"
+            );
+            $statement->bindValue(':paramValue', $params[2], \PDO::PARAM_STR);
+            $statement->bindValue(':cfg_nagios_id', $nagiosId, \PDO::PARAM_INT);
+            $statement->execute();
```
Comment (a-launois) [inline www/class/centreon-clapi/centreonEngineCfg.class.php:545]:
```suggestion
                "UPDATE cfg_nagios_logger SET `{$params[1]}` = :paramValue WHERE cfg_nagios_id = :cfgNagiosId"
            );
            $statement->bindValue(':paramValue', $params[2], \PDO::PARAM_STR);
            $statement->bindValue(':cfgNagiosId', $nagiosId, \PDO::PARAM_INT);
            $statement->execute();
```

File: www/install/php/Update-22.04.0-beta.1.php [hunk 12]
```diff
@@ -64,6 +64,66 @@
         REFERENCES `contact` (`contact_id`) ON DELETE CASCADE)"
     );
 
+    $errorMessage = "Unable to create table 'cfg_nagios_logger'";
+    $pearDB->query(
+        "CREATE TABLE IF NOT EXISTS `cfg_nagios_logger` (
```
Comment (a-launois) [inline www/install/php/Update-22.04.0-beta.1.php:69]: Move create in sql file

File: www/include/configuration/configNagios/DB-Func.php [hunk 13]
```diff
@@ -244,1992 +244,407 @@ function calculateBitwise($list)
...
-            . "',  ";
-    } else {
-        $rq .= "service_perfdata_file_processing_interval = NULL, ";
-    }
+/**
+ * @param CentreonDB $pearDB
+ * @param array $ret
+ * @param int $nagiosId
+ */
+function insertLoggerV2Cfg(CentreonDB $pearDB, array $ret, int $nagiosId): void
```
Comment (a-launois) [inline www/include/configuration/configNagios/DB-Func.php:445]: change $ret for $data when possible for more clarity

File: www/class/centreon-clapi/centreonEngineCfg.class.php [hunk 14]
```diff
@@ -212,11 +212,28 @@ public function initUpdateParameters($parameters)
...
+                'log_level_eventbroker',
+                'log_level_external_command',
+                'log_level_commands',
+                'log_level_downtimes',
+                'log_level_comments',
+                'log_level_macros',
+                'log_level_process',
+                'log_level_runtime',
+            ];
+            $setUpdateParams = true;
```
Comment (a-launois) [inline www/class/centreon-clapi/centreonEngineCfg.class.php:230]:
```suggestion
            $canUpdateParams = true;
```

**Review by callapa — CHANGES_REQUESTED** [review]

**Review by callapa — DISMISSED** [review]

File: www/install/php/Update-22.04.0-beta.1.php [hunk 15]
```diff
@@ -103,6 +103,7 @@
         $pearDB->query('ALTER TABLE `cfg_centreonbroker` ADD `bbdo_version` VARCHAR(50) DEFAULT "3.0.0"');
     }
 
+    // Add custom_configuration to provider configurations
```
Comment (a-launois) [inline www/install/php/Update-22.04.0-beta.1.php:106]:
```suggestion
```

File: www/install/createTables.sql [hunk 16]
```diff
@@ -2447,6 +2445,30 @@ CREATE TABLE `contact_password` (
...
+  `id` int(11) UNSIGNED NOT NULL AUTO_INCREMENT,
+  `cfg_nagios_id` int(11) NOT NULL,
+  `log_v2_logger` enum('file', 'syslog') DEFAULT 'file',
+  `log_level_functions` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'err',
+  `log_level_config` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'info',
+  `log_level_events` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'info',
+  `log_level_checks` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'info',
+  `log_level_notifications` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'err',
+  `log_level_eventbroker` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'err',
+  `log_level_external_command` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'err',
```
Comment (a-launois) [inline www/install/createTables.sql:2458]:
```suggestion
  `log_level_external_command` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'info',
```

File: www/install/sql/centreon/Update-DB-22.04.0-beta.1.sql [hunk 17]
```diff
@@ -1 +1,24 @@
...
+    `id` int(11) UNSIGNED NOT NULL AUTO_INCREMENT,
+    `cfg_nagios_id` int(11) NOT NULL,
+    `log_v2_logger` enum('file', 'syslog') DEFAULT 'file',
+    `log_level_functions` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'err',
+    `log_level_config` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'info',
+    `log_level_events` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'info',
+    `log_level_checks` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'info',
+    `log_level_notifications` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'err',
+    `log_level_eventbroker` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'err',
+    `log_level_external_command` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'err',
```
Comment (a-launois) [inline www/install/sql/centreon/Update-DB-22.04.0-beta.1.sql:13]:
```suggestion
    `log_level_external_command` enum('trace', 'debug', 'info', 'warning', 'err', 'critical', 'off') DEFAULT 'info',
```

File: www/install/var/baseconf/centreon-engine.sql [hunk 18]
```diff
@@ -1,8 +1,11 @@
...
+
+INSERT INTO `cfg_nagios_logger` (`cfg_nagios_id`, `log_v2_logger`, `log_level_functions`, `log_level_config`, `log_level_events`, `log_level_checks`, `log_level_notifications`, `log_level_eventbroker`, `log_level_external_command`, `log_level_commands`, `log_level_downtimes`, `log_level_comments`, `log_level_macros`, `log_level_process`, `log_level_runtime`) VALUES
+(1, 'file', 'err', 'info', 'info', 'info', 'err', 'err', 'err', 'err', 'err', 'err', 'err', 'info', 'err');
```
Comment (a-launois) [inline www/install/var/baseconf/centreon-engine.sql:8]:
```suggestion
(1, 'file', 'err', 'info', 'info', 'info', 'err', 'info', 'err', 'err', 'err', 'err', 'err', 'info', 'err');
```

File: src/CentreonRemote/Domain/Resources/DefaultConfig/CfgNagiosLogger.php [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+        return [
+            'cfg_nagios_id' => 1,
+            'log_v2_logger' => 'file',
+            'log_level_functions' => 'err',
+            'log_level_config' => 'info',
+            'log_level_events' => 'info',
+            'log_level_checks' => 'info',
+            'log_level_notifications' => 'err',
+            'log_level_eventbroker' => 'err',
+            'log_level_external_command' => 'err',
```
Comment (a-launois) [inline src/CentreonRemote/Domain/Resources/DefaultConfig/CfgNagiosLogger.php:45]:
```suggestion
            'log_level_external_command' => 'info',
```

File: tests/clapi_export/clapi-configuration.txt [hunk 20]
```diff
@@ -553,6 +551,21 @@ ENGINECFG;setparam;Centreon Engine Central;cfg_file;centengine.cfg
...
 ENGINECFG;setparam;Centreon Engine Central;broker_module;/usr/lib64/centreon-engine/externalcmd.so|/usr/lib64/nagios/cbmod.so /etc/centreon-broker/central-module.json
+ENGINECFG;setparam;Centreon Engine Central;logger_version;log_v2_enabled
+ENGINECFG;setparam;Centreon Engine Central;log_v2_logger;file
+ENGINECFG;setparam;Centreon Engine Central;log_level_functions;err
+ENGINECFG;setparam;Centreon Engine Central;log_level_config;info
+ENGINECFG;setparam;Centreon Engine Central;log_level_events;info
+ENGINECFG;setparam;Centreon Engine Central;log_level_checks;info
+ENGINECFG;setparam;Centreon Engine Central;log_level_notifications;err
+ENGINECFG;setparam;Centreon Engine Central;log_level_eventbroker;err
+ENGINECFG;setparam;Centreon Engine Central;log_level_external_command;err
```
Comment (a-launois) [inline tests/clapi_export/clapi-configuration.txt:562]:
```suggestion
ENGINECFG;setparam;Centreon Engine Central;log_level_external_command;info
```

File: www/class/centreon-config/centreonMainCfg.class.php [hunk 21]
```diff
@@ -75,6 +75,24 @@ class CentreonMainCfg
...
+    /** @var string[] */
+    private $loggerDefaultCfg = [
+        'log_v2_logger' => 'file',
+        'log_level_functions' => 'err',
+        'log_level_config' => 'info',
+        'log_level_events' => 'info',
+        'log_level_checks' => 'info',
+        'log_level_notifications' => 'err',
+        'log_level_eventbroker' => 'err',
+        'log_level_external_command' => 'err',
```
Comment (a-launois) [inline www/class/centreon-config/centreonMainCfg.class.php:87]:
```suggestion
        'log_level_external_command' => 'info',
```

**Review by adr-mo — DISMISSED** [review]

File: www/class/centreon-config/centreonMainCfg.class.php [hunk 22]
```diff
@@ -75,6 +75,24 @@ class CentreonMainCfg
         33554432 => 'Command data'
     ];
 
+    /** @var string[] */
```
Comment (kduret) [inline www/class/centreon-config/centreonMainCfg.class.php:78]:
```suggestion
    /** @var array<string,string> */
```

File: www/class/centreon-config/centreonMainCfg.class.php [hunk 23]
```diff
@@ -198,6 +214,16 @@ public function getDefaultMainCfg()
         return $this->aInstanceDefaultValues;
     }
 
+    /**
+     * Get default engine logger values
+     *
+     * @param string[]
```
Comment (kduret) [inline www/class/centreon-config/centreonMainCfg.class.php:220]:
```suggestion
     * @param array<string,string>
```

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 24]
```diff
@@ -191,19 +190,105 @@
...
+        <td class="FormRowField"><img class="helpTooltip" name="max_debug_file_size">&nbsp;{$form.max_debug_file_size.label}
+        </td>
+        <td class="FormRowValue">{$form.max_debug_file_size.html}&nbsp;{$Bytes}</td>
+      </tr>
+      <!-- LOGGER V2 OPTIONS -->
+      <tr class="list_one logger_v2_option">
+        <td class="FormRowField">{$form.log_v2_logger.label}</td>
+        <td class="FormRowValue">{$form.log_v2_logger.html}</td>
+      </tr>
+      <tr class="list_one logger_v2_option">
```
Comment (kduret) [inline www/include/configuration/configNagios/formNagios.ihtml:239]:
```suggestion
      <tr class="list_two logger_v2_option">
```

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 25]
```diff
@@ -191,19 +190,105 @@
...
+        <td class="FormRowField">{$form.log_level_checks.label}</td>
+        <td class="FormRowValue">{$form.log_level_checks.html}</td>
+      </tr>
+      <tr class="list_one logger_v2_option">
+        <td class="FormRowField">{$form.log_level_commands.label}</td>
+        <td class="FormRowValue">{$form.log_level_commands.html}</td>
+      </tr>
+      <tr class="list_one logger_v2_option">
```
Comment (kduret) [inline www/include/configuration/configNagios/formNagios.ihtml:247]:
```suggestion
      <tr class="list_two logger_v2_option">
```

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 26]
```diff
@@ -191,19 +190,105 @@
...
+        <td class="FormRowField">{$form.log_level_comments.label}</td>
+        <td class="FormRowValue">{$form.log_level_comments.html}</td>
+      </tr>
+      <tr class="list_one logger_v2_option">
+        <td class="FormRowField">{$form.log_level_config.label}</td>
+        <td class="FormRowValue">{$form.log_level_config.html}</td>
+      </tr>
+      <tr class="list_one logger_v2_option">
```
Comment (kduret) [inline www/include/configuration/configNagios/formNagios.ihtml:255]:
```suggestion
      <tr class="list_two logger_v2_option">
```

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 27]
```diff
@@ -191,19 +190,105 @@
...
+        <td class="FormRowField">{$form.log_level_downtimes.label}</td>
+        <td class="FormRowValue">{$form.log_level_downtimes.html}</td>
+      </tr>
+      <tr class="list_one logger_v2_option">
+        <td class="FormRowField">{$form.log_level_eventbroker.label}</td>
+        <td class="FormRowValue">{$form.log_level_eventbroker.html}</td>
+      </tr>
+      <tr class="list_one logger_v2_option">
```
Comment (kduret) [inline www/include/configuration/configNagios/formNagios.ihtml:263]:
```suggestion
      <tr class="list_two logger_v2_option">
```

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 28]
```diff
@@ -191,19 +190,105 @@
...
+        <td class="FormRowField">{$form.log_level_events.label}</td>
+        <td class="FormRowValue">{$form.log_level_events.html}</td>
+      </tr>
+      <tr class="list_one logger_v2_option">
+        <td class="FormRowField">{$form.log_level_external_command.label}</td>
+        <td class="FormRowValue">{$form.log_level_external_command.html}</td>
+      </tr>
+      <tr class="list_one logger_v2_option">
```
Comment (kduret) [inline www/include/configuration/configNagios/formNagios.ihtml:271]:
```suggestion
      <tr class="list_two logger_v2_option">
```

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 29]
```diff
@@ -191,19 +190,105 @@
...
+        <td class="FormRowField">{$form.log_level_functions.label}</td>
+        <td class="FormRowValue">{$form.log_level_functions.html}</td>
+      </tr>
+      <tr class="list_one logger_v2_option">
+        <td class="FormRowField">{$form.log_level_macros.label}</td>
+        <td class="FormRowValue">{$form.log_level_macros.html}</td>
+      </tr>
+      <tr class="list_one logger_v2_option">
```
Comment (kduret) [inline www/include/configuration/configNagios/formNagios.ihtml:279]:
```suggestion
      <tr class="list_two logger_v2_option">
```

File: www/include/configuration/configNagios/formNagios.ihtml [hunk 30]
```diff
@@ -191,19 +190,105 @@
...
+        <td class="FormRowField">{$form.log_level_notifications.label}</td>
+        <td class="FormRowValue">{$form.log_level_notifications.html}</td>
+      </tr>
+      <tr class="list_one logger_v2_option">
+        <td class="FormRowField">{$form.log_level_process.label}</td>
+        <td class="FormRowValue">{$form.log_level_process.html}</td>
+      </tr>
+      <tr class="list_one logger_v2_option">
```
Comment (kduret) [inline www/include/configuration/configNagios/formNagios.ihtml:287]:
```suggestion
      <tr class="list_two logger_v2_option">
```

File: www/include/configuration/configNagios/formNagios.php [hunk 31]
```diff
@@ -394,6 +412,93 @@ function ($macro) {
...
+            $key,
+            '&nbsp;',
+            $val,
+            array(
+                "id" => "debug" . $key,
+                "onClick" => "unCheckAllAndNaught('debug-level');",
+                'class' => 'debug-level'
+            )
+        );
+    }
```
Comment (kduret) [inline www/include/configuration/configNagios/formNagios.php:498]:
```suggestion
    $debugCheck[] = $form->createElement(
        'checkbox',
        $key,
        '&nbsp;',
        $val,
        [
            "id" => "debug" . $key,
            "onClick" => in_array($key, ['-1', '0'], true)
                ? "unCheckOthers('debug-level', this.name);"
                : "unCheckAllAndNaught('debug-level');",
            'class' => 'debug-level'
        ]
    );
```

File: www/include/configuration/configNagios/help.php [hunk 32]
```diff
@@ -155,15 +155,6 @@
     . "use_retained_program_state is enabled), you'll have to use the appropriate external command "
     . "or change it via the web interface. Option is enabled by default."
 );
-$help["log_rotation_method"] = dgettext(
-    "help",
-    "This is the rotation method that you would like Monitoring Engine to use for your log file."
```
Comment (kduret) [inline www/include/configuration/configNagios/help.php:160]: remove it from translation file

File: www/include/configuration/configNagios/help.php [hunk 33]
```diff
@@ -155,15 +155,6 @@
...
-);
-$help["log_archive_path"] = dgettext(
-    "help",
-    "This is the directory where Monitoring Engine should place log files that have been rotated. "
-    . "This option is ignored if you choose to not use the log rotation functionality."
```
Comment (kduret) [inline www/include/configuration/configNagios/help.php:165]: remove it from translation file

File: www/include/configuration/configNagios/help.php [hunk 34]
```diff
@@ -870,11 +861,6 @@
     "automatically be deleted. This helps ensure your disk space usage doesn't get out of control when debugging " .
     "Monitoring Engine."
 );
-$help["daemon_dumps_core"] = dgettext(
-    "help",
-    "This option allows dumping core in case a segmentation fault occurs. Warning: Make sure that server has " .
-    "sufficient disk space for these dumps (ulimit). This option is discarded when using Centreon Engine."
```
Comment (kduret) [inline www/include/configuration/configNagios/help.php:876]: remove it from translation file

File: src/CentreonRemote/Domain/Service/ConfigurationWizard/ServerConnectionConfigurationService.php [hunk 35]
```diff
@@ -153,6 +154,8 @@ protected function insertConfigNagios($serverID): int
     {
         $configID = $this->insertWithAdapter('cfg_nagios', CfgNagios::getConfiguration($this->name, $serverID));
 
+        $this->insertWithAdapter('cfg_nagios_logger', CfgNagiosLogger::getConfiguration($configID));
```
Comment (kduret) [inline src/CentreonRemote/Domain/Service/ConfigurationWizard/ServerConnectionConfigurationService.php:157]:
```suggestion
        $this->insertWithAdapter('cfg_nagios_logger', new (CfgNagiosLogger($configID)));
```

File: src/CentreonRemote/Domain/Resources/RemoteConfig/CfgNagiosLogger.php [hunk 36]
```diff
@@ -0,0 +1,43 @@
...
+
+namespace CentreonRemote\Domain\Resources\RemoteConfig;
+
+use CentreonRemote\Domain\Resources\DefaultConfig\CfgNagiosLogger as DefaultLoggerCfg;
+
+/**
+ * Get broker configuration template
+ */
+class CfgNagiosLogger
+{
```
Comment (kduret) [inline src/CentreonRemote/Domain/Resources/RemoteConfig/CfgNagiosLogger.php:30]: add construct with id

**Review by callapa — DISMISSED** [review]

**Review by callapa — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**